### PR TITLE
Feature/529: FastApi + Django application templates

### LINKED ISSUE #529: Create Django application template
Create a new application template to support development of Django based applications with contract first approach using OpenAPI.

The new application template should support:
- generating backend and frontend
- integration with keycloak users
- FastAPI server
- Django
- pydantic models
- uvicorn (async server)

### PR DESCRIPTION
**Pull request by zsinnema**
Closes #529 

Implemented solution: application template + docker image

How to test this PR: .createe a neww applicaction using gthe -t django-app template

### Sanity checks:
- [x] The pull request is explicitly linked to the relevant issue(s)
- [x] The issue is well described: clearly states the problem and the general proposed solution(s)
- [x] From the issue and the current PR it is explicitly stated how to test the current change
- [x] The labels in the issue set the scope and the type of issue (bug, feature, etc.)
- [x] All the automated test checks are passing
- [x] All the linked issues are included in one milestone
- [x] All the linked issues are in the Review/QA column of the [board](https://app.zenhub.com/workspaces/cloud-harness-5fdb203b7e195b0015a273d7/board)
- [x] All the linked issues are assigned

### Breaking changes (select one):
- [x] The present changes do not change the preexisting api in any way
- [ ] This PR and the issue are tagged as a `breaking-change`

### Possible deployment updates issues (select one):
- [x] There is no reason why deployments based on CloudHarness may break after the current update
- [ ] This PR and the issue are tagged as `alert:deployment`

### Test coverage (select one):
- [ ] Tests for the relevant cases are included in this pr
- [x] The changes included in this pr are out of the current test coverage scope

### Documentation (select one):
- [x ] The documentation has been updated to match the current changes
- [ ] The changes included in this PR are out of the current documentation scope

### Nice to have (if relevant):
- [ ] Screenshots of the changes
- [ ] Explanatory video/animated gif

### CODE REVIEW

File: docs/applications/harness-application.md [hunk 1]
```diff
@@ -17,14 +17,16 @@ positional arguments:
 optional arguments:
   -h, --help            show this help message and exit
   -t TEMPLATES, --template TEMPLATES
-                        Add a template name. Available templates: - base (backend flask app based on openapi, always included) - webapp (webapp including backend and frontend) - db-postgres - db-neo4j - db-mongo
+                        Add a template name. Available templates: - base (always included) - server (backend flask app based on openapi) - webapp (webapp including backend and frontend) - db-postgres - db-neo4j - db-mongo - django-app (fastapi django backend based on openapi)
```
Comment (filippomc) [inline docs/applications/harness-application.md:20]: does the django app include also a frontend webapp? I'm also for renaming the server name to be flask something to be clear, no need to keep the old name

File: docs/applications/harness-application.md [hunk 2]
```diff
@@ -42,4 +44,10 @@ optional arguments:
...
 * `db-neo4j`- [Neo4J](https://neo4j.com/), a graph database
-* `db-mongo` - [MongoDB](https://www.mongodb.com/), a NoSQL document-based database
\ No newline at end of file
+* `db-mongo` - [MongoDB](https://www.mongodb.com/), a NoSQL document-based database
+
+### Django
+* It consists of a single backend, a Python [FastAPI](https://fastapi.tiangolo.com/) application. 
+* The [FastAPI code generator](https://github.com/koxudaxi/fastapi-code-generator) maps the OpenAPI definition to FastAPI routing.
+* The [Django framework](https://www.djangoproject.com/) encourages rapid development and clean, pragmatic design.
+* Per default, [Uvicorn](https://www.uvicorn.org/) serves the FastAPI app with 2 workers. Depending on the application requirements, you can update the number of workers.
```
Comment (filippomc) [inline docs/applications/harness-application.md:53]: Have to check whether the cloudharness debug still works. It's now based on replacing the gunicorn command with something like `python __main__.py` but I'm not sure the detection still works here

File: docs/applications/harness-application.md (hunk 1 above)
Comment (zsinnema) [reply docs/applications/harness-application.md:20]: yes the django app add the webapp automatically

File: docs/applications/harness-application.md (hunk 2 above)
Comment (zsinnema) [reply docs/applications/harness-application.md:53]: yes this still works as expected :-)

File: docs/applications/harness-application.md (hunk 1 above)
Comment (zsinnema) [reply docs/applications/harness-application.md]: I renamed server to flask-server

**Review by filippomc — APPROVED** [review]

**Pull request merged**